app/domain/authentication/authn_gce/validate_resource_restrictions_configuration.rb

```diff
@@ -0,0 +1,49 @@
+module Authentication
+  module AuthnGce
+
+    # This class is responsible for validating resource restrictions configuration.
+    # the allowed configuration is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    ValidateResourceRestrictionsConfiguration = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions permitted_constraints)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGce::ValidatingResourceRestrictionsConfiguration.new)
+        validate_required_restrictions_exist
+        validate_constraints_are_permitted
+        validate_restrictions_values_exist
+        @logger.debug(LogMessages::Authentication::AuthnGce::ValidatedResourceRestrictionsConfiguration.new)
```

Comment:
@orenbm were these logs approved by shuli?

Reply:
no, still waiting for an approval